src/pal/src/thread/process.cpp

```diff
@@ -1503,6 +1539,11 @@ class PAL_RuntimeStartupHelper
     DWORD m_threadId;
     HANDLE m_threadHandle;
     DWORD m_processId;
+#ifdef __APPLE__    
+    CharString<MAX_APPLICATION_GROUP_ID_LENGTH> m_applicationGroupId;
```

Comment:
Since there is a small fixed limit on the application group it, can you please store this as a char[] instead?